var/spack/repos/builtin/packages/placement-algorithm/package.py

```diff
@@ -13,27 +13,29 @@ class PlacementAlgorithm(PythonPackage):
     git      = "git@bbpgitlab.epfl.ch:nse/placement-algorithm.git"
 
     version('develop', branch='master')
-    version('2.1.2', tag='placement-algorithm-v2.1.2')
-    version('2.1.1', tag='placement-algorithm-v2.1.1')
-    version('2.1.0', tag='placement-algorithm-v2.1.0')
-    version('2.0.10', tag='placement-algorithm-v2.0.10')
-    version('2.0.8', tag='placement-algorithm-v2.0.8')
+    version('2.2.0', tag='placement-algorithm-v2.2.0')
+    version('2.1.4', tag='placement-algorithm-v2.1.4')
 
     build_directory = 'python'
 
     depends_on('py-setuptools', type=('build', 'run'))
 
-    depends_on('py-lxml', type='run')
-    depends_on('py-numpy', type='run')
-    depends_on('py-pandas', type='run')
+    depends_on('py-lxml@4.0:', type='run')
+    depends_on('py-numpy@1.8:', type='run')
+    depends_on('py-pandas@0.19:', type='run')
     depends_on('py-six', type='run')
 
-    depends_on('py-morphio@2.0.5:', type='run')
-    depends_on('py-morph-tool@0.1.3:', type='run')
-    depends_on('py-mpi4py@2.0:', type='run')
+    depends_on('py-morphio@3.0:3.999', type='run')
+    depends_on('py-morph-tool@2.9.0:', type='run', when='@2.2.0:')
+    depends_on('py-morph-tool@0.1.3:2.8.99', type='run', when='@:2.1.99')
+    depends_on('py-neurom@3.0:3.999', type='run', when='@2.2.0:')
+    depends_on('py-neurom@2.0:2.99', type='run', when='@:2.1.99')
+    depends_on('py-neuroc@:0.2.7', type='run', when='@:2.1.99')
+    depends_on('py-neuroc@0.2.8:0.999', type='run', when='@2.2.0:')
+    depends_on('py-mpi4py@3.0.3:', type='run')
     depends_on('py-tqdm@4.0:', type='run')
     depends_on('py-voxcell@2.5:2.6.99', when='@:2.0.99', type='run')
-    depends_on('py-voxcell@2.7:', when='@2.1.0:', type='run')
-    depends_on('py-dask@2.15:', when='@2.0.12:', type='run')
+    depends_on('py-voxcell@2.7:3.99', when='@2.1.0:', type='run')
+    depends_on('py-dask+distributed+bag@2.15:', when='@2.0.12:', type='run')
```

Comment:
The requirement `depends_on('py-region-grower@0.1.5:0.1.99', type='run', when='@:2.1.99')` cannot be satisfied using the remaining versions of `py-region-grower` (0.3.0, 0.2.4) so I'm surprised that https://bbpcode.epfl.ch/ci/blue/organizations/jenkins/hpc.spack-deployment.pull-requests/detail/hpc.spack-deployment.pull-requests/2419/pipeline/84/ didn't fail.
However, https://bbpcode.epfl.ch/ci/blue/rest/organizations/jenkins/pipelines/hpc.spack-deployment.pull-requests/runs/2419/nodes/84/steps/89/log/?start=0 reports some warnings and errors:
```
Warning: Missing a source id for py-region-grower@0.1.99
Error: Can't extrapolate a URL for version 0.1.99 because package py-region-grower defines no URLs
```
and `placement-algorithm@2.1.4` wasn't built.

Moreover, I see that `py-region-grower-0.2.4` was built successfully and can be loaded, while `py-region-grower-0.3.3` is incomplete. Compare for example:
```
/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1267/deploy/applications/2021-08-30/linux-rhel7-x86_64/gcc-9.3.0/py-region-grower-0.2.4-cwfbqj
/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/pulls/1267/deploy/applications/2021-08-30/linux-rhel7-x86_64/gcc-9.3.0/py-region-grower-0.3.0-2zlvey
```
Maybe it's better to clean the workdir, fix this dependency and start the build again?

Reply:
> Maybe it's better to clean the workdir

I don't see any functionality in on my web page for that. Please advice.

> fix this dependency

Do you mean to return back the required version of `py-region-grower`?

Reply:
@asanin-epfl I don't think there is a functionality to delete the directory.
Bringing back the old version of `py-region-grower` as you did is the right thing to do if required by `placement-algorithm@2.1.4`.

@matz-e could you please clean the build dir used by jenkins for this PR?

Reply:
> @matz-e could you please clean the build dir used by jenkins for this PR?

Done!